include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -79,6 +79,23 @@ enum class charge_distribution_history
     NEGLECT
 };
 
+/**
+ * An enumeration of modes to specifying if the charge index should be recomputed fully.
+ */
+enum class charge_index_recomputation
+{
+    /**
+     * The charge index is recomputed from scratch since we don't know if the the charge index was incremented after the
+     * last charge index computation.
+     */
+    FROM_SCRATCH,
+    /**
+     * The charge index is recomputed with the leading zeroes ignored. We can apply this optimization if we know that
+     * the charge index was incremented after the last charge index computation.
```

Comment:
```suggestion
was changed after the last charge index computation.
```
is this possible, correct?